api/swagger.json

```diff
@@ -0,0 +1,1304 @@
+      "externalDocs": {
+        "description": "Find out more",
+        "url": "http://chapter.io"
+      }
+    },
+    {
+      "name": "user",
+      "description": "Everything about your Users",
+      "externalDocs": {
+        "description": "Find out more",
+        "url": "http://chapter.io"
+      }
+    }
+  ],
+  "paths": {
+    "/locations": {
+      "post": {
+        "tags": [
+          "location"
+        ],
+        "summary": "Add a new location to the map",
+        "description": "",
+        "operationId": "addLocation",
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/Location"
+              }
+            }
+          },
+          "description": "Location coordinates",
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "Location response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Location"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "Not authorized"
+          },
+          "400": {
+            "description": "Invalid Location object"
+          },
+          "default": {
+            "description": "unexpected error",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "location"
+        ],
+        "summary": "Finds Locations by city, country etc",
+        "description": "Multiple status values can be provided as query params",
+        "operationId": "findLocations",
+        "parameters": [
+          {
+            "name": "city",
+            "in": "query",
+            "description": "City value that needs to be considered for filter",
+            "required": false,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "country",
+            "in": "query",
+            "description": "Country value that needs to be considered for filter",
+            "required": false,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "zip",
+            "in": "query",
+            "description": "Zip Code that need to be considered for filter",
+            "required": false,
+            "schema": {
+              "type": "integer"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
```

Comment:
Do we want to have a wrapping object that includes result metadata, with e.g. a "data" or "locations" key that contains the actual result set? That allows for e.g. adding hypermedia controls to the API later without a BC break, which you'd have if you started with an unwrapped array.